Makefile

```diff
@@ -24,11 +33,17 @@ build: autogen
 	$(COMMONENVVAR) $(BUILDENVVAR) go build -ldflags '-w' -o bin/kube-scheduler cmd/main.go
 
 .PHONY: local-image
-local-image: autogen
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags '-w' -o ./build/kube-scheduler cmd/main.go
-	chmod +x ./build/kube-scheduler
-	docker build -t localhost:$(REG_PORT)/scheduler-plugins:latest ./build
-	rm ./build/kube-scheduler
+local-image: clean
+	docker build -t $(LOCAL_REGISTRY)/$(LOCAL_IMAGE) .
+
+.PHONY: release-image
+release-image: clean
+	docker build -t $(RELEASE_REGISTRY)/$(RELEASE_IMAGE) .
+
+.PHONY: push-release-image
+push-release-image: release-image
```

Comment:
Two questions:

1. When this will be triggered?
1. When the staging image will be populated to k8s.gcr.io?

Reply:
> Two questions:
> 
> 1. When this will be triggered?

I am proposing triggering this every time a commit is pushed to the master branch and release branches. And also when tags are created. This will push to the staging registry not the production(k8s.gcr.io) registry.

See the below lines from the Prow job definition.
https://github.com/kubernetes/test-infra/pull/19215/files#diff-63bd6057dfb31cdf6c9e35d03e35eec5R12-R16

> 2. When the staging image will be populated to k8s.gcr.io?

All container image promotions from the staging registry to the production(k8s.gcr.io) registry are done using this process:

https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io#image-promoter

Reply:
> I am proposing triggering this every time a commit is pushed to the master branch and release branches. And also when tags are created.

SG. I noticed sig-storage is also following this pattern: https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-sig-storage.yaml